### PR TITLE
cap retries on 409 when creating session

### PR DESCRIPTION
**Pull request by scottdallamura**

### CODE REVIEW

File: src/agent/api/messagelistener.ts [hunk 1]
```diff
@@ -79,13 +83,28 @@ export class MessageListener extends events.EventEmitter {
...
+            
             if (err) {
                 onError(new Error('Could not create an agent session.  Retrying in ' + QUEUE_RETRY_DELAY/1000 + ' sec'));
                 onError(err);
-                setTimeout(() => {
-                        this.start(callback, onError);
-                    }, QUEUE_RETRY_DELAY);
+                
+                // retry 409 (session already exists) a few times
+                if (statusCode == 409) {
```
Comment (jpricketMSFT) [inline src/agent/api/messagelistener.ts]: I am not sure how the logging works for the xplat agent, but in the .net agent we log each attempt so you can easily see that it was retried.

**Comment by jpricketMSFT** [comment]
Changes look good! I would suggest adding some logging, but that's all.

File: src/agent/api/messagelistener.ts (hunk 1 above)
Comment (scottdallamura) [inline src/agent/api/messagelistener.ts]: We still do that on lines 88 and 89.

Comment (bryanmacfarlane) [inline src/agent/api/messagelistener.ts]: Writing to the console shows up in the OSX services log (or the console if running interactively)

Comment (bryanmacfarlane) [inline src/agent/api/messagelistener.ts]: Also note that the message listener implements event emmitter.  So, you can implement a giving up (think of better name) event then the agent can handle and put in it's log as well

File: src/agent/api/messagelistener.ts [hunk 2]
```diff
@@ -79,13 +83,28 @@ export class MessageListener extends events.EventEmitter {
...
+                    if (this._sessionRetryCount++ < MAX_SESSION_RETRIES) {
```
Comment (bryanmacfarlane) [inline src/agent/api/messagelistener.ts]: Shouldn't we also reset sessionRetryCount to 0 on success??

**Pull request merged**